### PR TITLE
kvserver: log when acquiring lease as a draining node

### PR DESCRIPTION
**Pull request by aayushshah15**
Generally, draining nodes are not allowed to acquire leases. Additionally,
replicas that are not Raft leaders are not allowed to acquire leases. This
means that draining nodes have to be permitted to acquire leases for replicas
that they are the Raft leaders for (otherwise, we risk a deadlock since no
replicas will be allowed to acquire the lease).

This commit adds a verbose log event when a draining store acquires a lease
because it is the Raft leader.

Release note: none

Jira issue: CRDB-14751

### CODE REVIEW

**Review by shralex — APPROVED** [review]

File: pkg/kv/kvserver/replica_range_lease.go [hunk 1]
```diff
@@ -772,13 +772,26 @@ func (r *Replica) requestLeaseLocked(
...
+			return r.mu.pendingLeaseRequest.newResolvedHandle(
+				roachpb.NewError(
+					newNotLeaseHolderError(
+						roachpb.Lease{}, r.store.StoreID(), r.mu.state.Desc,
+						"refusing to take the lease; node is draining",
+					),
+				),
+			)
+		}
+		log.VEventf(
```
Comment (nvanbenschoten) [inline pkg/kv/kvserver/replica_range_lease.go:792]: This won't usually end up in the logs. Do we want to promote it to `log.Info` so that it will?

**Review by nvb — APPROVED** [review]

File: pkg/kv/kvserver/replica_range_lease.go (hunk 1 above)
Comment (aayushshah15) [reply pkg/kv/kvserver/replica_range_lease.go]: Done.

**Comment by aayushshah15** [comment]
TFTR 

bors r+

**Comment by aayushshah15** [comment]
blathers backport 21.2 22.1

**Pull request merged**